core/storage/base_model/gae_models.py

```diff
@@ -215,13 +215,15 @@ def get_field_names_for_takeout(cls):
         return {}
 
     @classmethod
-    def get(cls, entity_id, strict=True):
+    def get(cls, entity_id, strict=True, return_deleted=False):
         """Gets an entity by id.
 
         Args:
             entity_id: str. The entity id.
             strict: bool. Whether to fail noisily if no entity with the given id
                 exists in the datastore. Default is True.
+            return_deleted: bool. Whether to return entity even if it is marked
```

Comment:
This seems to conflate with ``strict`` a bit. I think it might be a bit cleaner to combine these into an enumeration of constants to describe the deletion retrieval mode. It seems that there are three cases that we care about:
- Throw if entity doesn't exist
- Return None if entity doesn't exist
- Return the deleted entity if it doesn't exist, and there's a deleted version

That being said, it seems a bit ick to let anyone retrieve a deleted model at any time--part of the reason why we delete stuff is to never retrieve it again. Can you provide some context why we have to use a deleted model retrieval for UserIdentifiersModel?

If it's absolutely necessary, and given this is the first time we've ever needed this, I think it would be preferable to isolate the support in specifically UserIdentifiersModel rather than opening this up as functionality for all models in Oppia. The latter can have unintended consequences since it effectively turns deleted models into "sort of not deleted anymore."

Reply:
In some cases, we use `get_by_id` (method provided by ndb) to retrieve even deleted models, but we are trying to remove the usage of `get_by_id` from the codebase (#11219). I agree with your proposed solutions, for now, I have implemented method directly in the `UserIdentifiersModel` and I will think about it more if we need to access deleted models from other places too.